opm/simulators/wells/GasLiftRuntime_impl.hpp

```diff
@@ -0,0 +1,798 @@
+/*
+  Copyright 2020 Equinor ASA.
+
+  This file is part of the Open Porous Media project (OPM).
+
+  OPM is free software: you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation, either version 3 of the License, or
+  (at your option) any later version.
+
+  OPM is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with OPM.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#include <opm/simulators/wells/StandardWell.hpp>
+#include <opm/parser/eclipse/EclipseState/Schedule/Schedule.hpp>
+#include <opm/parser/eclipse/EclipseState/Schedule/Well/Well.hpp>
+#include <opm/simulators/utils/DeferredLogger.hpp>
+#include <opm/parser/eclipse/EclipseState/Schedule/SummaryState.hpp>
+#include <opm/simulators/wells/WellStateFullyImplicitBlackoil.hpp>
+#include <opm/parser/eclipse/EclipseState/Schedule/GasLiftOpt.hpp>
+
+#include <optional>
+#include <string>
+
+template<typename TypeTag>
+Opm::GasLiftRuntime<TypeTag>::
+GasLiftRuntime(
+    const StdWell &std_well,
+    const Simulator &ebos_simulator,
+    const SummaryState &summary_state,
+    DeferredLogger &deferred_logger,
+    std::vector<double> &potentials,
+    const WellState &well_state,
+    const Well::ProductionControls &controls
+) :
+    controls_{controls},
+    deferred_logger_{deferred_logger},
+    ebos_simulator_{ebos_simulator},
+    potentials_{potentials},
+    std_well_{std_well},
+    summary_state_{summary_state},
+    well_state_{well_state},
+    debug{true},  // extra debugging output
+    debug_disable{false} // act as glift optimization is disabled
+{
```

Comment:
I think we should not output the debug info as default
Also do we need the posibility to hard-code disabling of the glift optimization anymore

Reply:
Thanks for the comment! I have updated the PR.

Reply:
Good